src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Usage/ProvideCorrectArgumentToEnumHasFlagTests.cs

```diff
@@ -137,9 +140,10 @@ public enum OtherEnum { A, }
 
     public void Method(MyEnum m)
     {
-        [|m.HasFlag(OtherEnum.A)|];
+        {|#0:m.HasFlag(OtherEnum.A)|};
     }
-}");
+}",
+                VerifyCS.Diagnostic(ProvideCorrectArgumentToEnumHasFlag.DifferentTypeRule).WithLocation(0).WithArguments("OtherEnum", "MyEnum"));
```

Comment:
I decided to report only one diagnostic here because when types are different we do not know which of the two types is the right one (it's most likely the one on which the method is called though) thus I wasn't sure on which type to base the check. Note that we could also go for 3 diagnostics in this case, the old one, one for the fact that `MyEnum` is not marked with `[Flags]` and one for the fact that `OtherEnum` is not marked with `[Flags]`. Let me know if you have any preference.

Reply:
Single diagnostic sounds reasonable.